src/NUnitTestAdapter/TestConverter.cs

```diff
@@ -199,6 +199,35 @@ private VSTestResult GetBasicResult(XmlNode resultNode)
             if (outputNode != null)
                 vsResult.Messages.Add(new TestResultMessage(TestResultMessage.StandardOutCategory, outputNode.InnerText));
 
+            var attachments = resultNode.SelectNodes("attachments/attachment");
+            if (attachments.Count > 0)
+            {
+                var attachmentSet = new AttachmentSet(new Uri(NUnitTestAdapter.ExecutorUri), "Attachments");
+                vsResult.Attachments.Add(attachmentSet);
+
+                foreach (XmlNode attachment in attachments)
+                {
+                    var path = attachment.SelectSingleNode("filePath")?.InnerText ?? string.Empty;
+                    var description = attachment.SelectSingleNode("description")?.InnerText;
+
+                    try
+                    {
+                        // We only support absolute paths since we dont lookup working directory here
+                        // any problem with path will throw an exception
+                        var fileUri = new Uri(path, UriKind.Absolute);
+                        attachmentSet.Attachments.Add(new UriDataAttachment(fileUri, description));
+                    }
+                    catch (UriFormatException ex)
+                    {
+                        _logger.Warning(string.Format("Ignoring attachment with path '{0}' due to problem with path: {1}", path, ex.Message));
```

Comment:
👍 I will use string interpolation, was unsure about coding standard for this project

Btw. I would be really nice if the readme at the root would be updated with some minor details such as requireing VS 2017 for development etc.  If was after writing this that I saw that the new sdk msbuild project was used and it was safe to use new C# functionality.